lib/teleterm/gateway/local_proxy_middleware.go

```diff
@@ -41,13 +43,20 @@ func (m *localProxyMiddleware) OnNewConnection(ctx context.Context, lp *alpn.Loc
 	m.log.Debug("Checking local proxy certs")
 
 	err := lp.CheckDBCerts(m.dbRoute)
-	if err != nil {
-		m.log.WithError(err).Debug("Gateway certificates have expired")
+	if err == nil {
+		return nil
+	}
+
+	// Return early and don't fire onExpiredCert if certs are invalid but not due to expiry.
+	if !errors.As(err, &x509.CertificateInvalidError{}) {
+		return trace.Wrap(err)
+	}
+
```

Comment:
I don't know how unorthodox this is in Go, but I return early if there's no error or if the error is not related to cert expiry. This way there's less indentation around the code which does the actual work of calling `onExpiredCert`

Reply:
I like early return - less "trailing off to the right" in complicated logic